src/app/utils.ts

```diff
@@ -212,3 +211,27 @@ export const onEscHandler = (element: React.RefObject<HTMLElement>, isEnabled: b
 export const useOnEsc = (element: React.RefObject<HTMLElement>, isEnabled: boolean, cb: () => void) => {
   React.useEffect(onEscHandler(element, isEnabled, cb), [element, isEnabled]);
 };
+
+export const useOnDOMEvent = (
+  element: React.RefObject<HTMLElement>['current'] | Window ,
+  isEnabled: boolean,
+  event: keyof HTMLElementEventMap,
+  cb: () => void
+) => {
+  React.useEffect(() => {
+    if (!element) { return; }
+
+    if (isEnabled) {
+      element.addEventListener(event, cb);
+    } else {
+      element.removeEventListener(event, cb);
+    }
+
+    return () => element.removeEventListener(event, cb);
+  }, [element, isEnabled, cb, event]);
+};
+
+export const useTimeout = (delay: number, cb: () => void, deps: any[]) => React.useEffect(() => {
```

Comment:
what is the type of the deps argument to useEffect? is it `any[]`?

Reply:
Its `ReadonlyArray<any>` but i've updated it. Not sure how to fix the linter errors about dependency list not being an array literal though.